code/modules/clothing/spacesuits/flightsuit.dm

```diff
@@ -492,6 +520,38 @@
 		losecontrol(move = FALSE)
 	crashing = FALSE
 
+/obj/item/device/flightpack/proc/door_hit(obj/structure/mineral_door/door)
```

Comment:
That's when you add new one. Cost of a function on atom/movable is negligible and makes this much easier to expand and change.